src/artm/core/data_loader.cc

```diff
@@ -114,18 +117,36 @@ void LocalDataLoader::AddBatch(const Batch& batch, bool invoke) {
     BatchHelpers::PopulateClassId(modified_batch.get());
   }
 
-  boost::uuids::uuid uuid = generation_->AddBatch(modified_batch);
+  if (invoke) {
+    auto time_start = boost::posix_time::microsec_clock::local_time();
+    for (;;) {
+      if (instance_->processor_queue()->size() < config.processor_queue_max_size()) break;
 
-  if (invoke)
-    instance_->batch_manager()->Add(uuid, std::string());
-}
+      boost::this_thread::sleep(boost::posix_time::milliseconds(kIdleLoopFrequency));
 
+      if (timeout >= 0) {
+        auto time_end = boost::posix_time::microsec_clock::local_time();
+        if ((time_end - time_start).total_milliseconds() >= timeout) return false;
+      }
+    }
+    auto pi = std::make_shared<ProcessorInput>();
+    pi->mutable_batch()->CopyFrom(*modified_batch);
+    boost::uuids::uuid uuid = boost::uuids::random_generator()();
+    pi->set_batch_uuid(boost::lexical_cast<std::string>(uuid));
+    instance_->batch_manager()->AddAndNext(BatchManagerTask(uuid, std::string()));
+    instance_->processor_queue()->push(pi);
+  } else {
+    generation_->AddBatch(modified_batch);
```

Comment:
Please, open a new issue 'support cache_theta flag in the online_processing_mode'. This is need in order to remove AddBatch() for non-online_processing_mode. As the result we will be able to clean-up things like MemoryGeneration and some other code.